crates/common_utils/src/id_type/global_id/payment.rs

```diff
@@ -17,11 +19,26 @@ pub struct GlobalPaymentId(super::GlobalId);
 // Database related implementations so that this field can be used directly in the database tables
 crate::impl_queryable_id_type!(GlobalPaymentId);
 
+#[derive(Debug, thiserror::Error, Clone, PartialEq, Eq)]
+pub enum GlobalPaymentIdError {
+    #[error("Failed to construct GlobalPaymentId")]
+    ConstructionError,
+}
+
 impl GlobalPaymentId {
     /// Get string representation of the id
     pub fn get_string_repr(&self) -> &str {
         self.0.get_string_repr()
     }
+
+    /// Generate a new GlobalPaymentId from a cell id
+    pub fn generate(cell_id: &str) -> error_stack::Result<Self, GlobalPaymentIdError> {
```

Comment:
we will always have a cell id in state.conf.cell_information. We sould use the typed value instead. So the generate function will never fail.